googleapiclient/http.py

```diff
@@ -91,12 +95,24 @@ def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
 
     try:
       resp, content = http.request(uri, method, *args, **kwargs)
-    except ssl.SSLError:
+    except (ssl.SSLError, TimeoutError) as ex:
       if retry_num == num_retries:
         raise
       else:
+        logging.warning('Encountered exception %s', repr(ex))
         continue
-    if resp.status < 500:
+    reason = ''
+    try:
+      if resp and resp.status == 403:
+        data = json.loads(content.decode('utf-8'))
+        reason = data['error']['errors'][0]['reason']
+        logging.warning('Encountered status %d with reason "%s"' %
+                        (resp.status, reason))
+    except Exception:
```

Comment:
@jonparrott The various cases where parsing failed, e.g. `content` wasn't populated, or `json.loads()` couldn't parse it, or it could parse it but `data['error']['errors'][0]['reason']` is an invalid path... And possibly other things I can't think of.

Is there a way to get a comprehensive list of exceptions that may be thrown? If not, I'd be risking failing the request due to a parsing error in the retry logic, which would be very bad... I'm fairly new to Python so I'd love to learn a better way! :)